src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -617,7 +617,14 @@ private static string Encode5987(StringSegment input)
                 //      ; token except ( "*" / "'" / "%" )
                 if (c > 0x7F) // Encodes as multiple utf-8 bytes
                 {
-                    var bytes = Encoding.UTF8.GetBytes(c.ToString());
+                    var startPos = i;
+                    while (i + 1 < input.Length && input[i + 1] > 0x7F)
+                    {
+                        i++;
+                    }
+
+                    var unicodePart = input.Value.Substring(startPos, i - startPos + 1);
```

Comment:
Can you make this span-based to avoid the intermediate allocation(s)?

Reply:
Good point, we can optimize allocations here if we have multiple unicode segments. Anyway, we still need to allocate a buffer to get bytes. Update is in progress.

Reply:
> need to allocate a buffer to get bytes.

Encoding has an overload that takes Span<byte> too, so no allocation needed. Either stackalloc or rent from the ArrayPool to avoid that allocation also.

Reply:
Used ArrayPool in this case as it applies better here and we can avoid stackalloc upfront.